pkgs/development/python-modules/granian/default.nix

```
@@ -39,6 +40,16 @@ buildPythonPackage rec {
     maturinBuildHook
   ];
 
+  buildInputs = [
+    (rust-jemalloc-sys.overrideAttrs ({ configureFlags, ... }: {
+      configureFlags = configureFlags ++ [
+        # otherwise import check fails with:
```

hydra doesn't show this issue, so it is easy for unknowing contributors to drop this fix
```suggestion
        # otherwise import check on aarch64-linux with 16k page size fails with:
```

---

Actually with "standard" `rust-jemalloc-sys` being injected the `cannot allocate memory in static TLS block` error happens on x64_64 too. That's a separate issue from the original bug that the `rust-jemalloc-sys` addition is fixing.

---

Then please document that rust-jemalloc-sys is added to buildInputs to address 16k pages

---

I think that got added in a tree wide to compile against our jenalloc, not the vendor one, and also fixes this issue. The other packages I looked at didn't have a comment either.

I am not sure about the override. Maybe seeing some env would be better?

---

The reason i ask for a comment is to document under what condition the input may be dropped

---

I don't think under any. Ideally we never want to link against vendored libraries unless absolutely necessary. That it fixes aarch64-linux is only a good side effect.

---

Also I only tested this on x86_64-linux where the import check also failed.